dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumApplicationScope.kt

```diff
@@ -65,4 +87,52 @@ internal class RumApplicationScope(
     }
 
     // endregion
+
+    @WorkerThread
+    private fun delegateToChildren(
+        event: RumRawEvent,
+        writer: DataWriter<Any>
+    ) {
+        val iterator = childScopes.iterator()
+        @Suppress("UnsafeThirdPartyFunctionCall") // next/remove can't fail: we checked hasNext
+        while (iterator.hasNext()) {
+            val result = iterator.next().handleEvent(event, writer)
+            if (result == null) {
+                iterator.remove()
+            }
+        }
+    }
+
+    @WorkerThread
+    private fun startNewSession(event: RumRawEvent, writer: DataWriter<Any>) {
+        val newSession = RumSessionScope(
+            this,
+            sdkCore,
+            samplingRate,
+            backgroundTrackingEnabled,
+            trackFrustrations,
+            firstPartyHostHeaderTypeResolver,
+            cpuVitalMonitor,
+            memoryVitalMonitor,
+            frameRateVitalMonitor,
+            sessionListener,
+            contextProvider,
+            true
+        )
+        childScopes.add(newSession)
+        if (event !is RumRawEvent.StartView) {
+            lastActiveViewScope?.let {
+                if (it.keyRef.get() != null) {
+                    // Restart the last active view
+                    val startViewEvent = RumRawEvent.StartView(
+                        key = it.keyRef.get()!!,
+                        name = it.name,
+                        attributes = it.attributes
+                    )
+                    newSession.handleEvent(startViewEvent, writer)
+                }
+            }
+        }
+        lastActiveViewScope = null
```

Comment:
this looks like a hack, can you please elaborate the purpose of this? I have concerns about mutable and public property `lastActiveViewScope`, looks like it is leaking encapsulation and the nesting concept.

Reply:
Yeah, this is the part I'm most concerned about as well.

The requirements are that when start a new session from a user action, it reactivates the last known view in thew new session.  The only good way I've found to do this is to reach down to get the currently active view in scope and save it until the new session is started.

The alternative would be either to have session scope keep its own copy of the currently active view and it's properties modified by events (which I worry could get out of sync), or broadcast up the relevant information on change, which also requires a store of the property. I'm open to either approach though I think the second is likely better.

But, `lastActiveViewScope` shouldn't be public either way.